### PR TITLE
Enable ROBOTOLOGY_USES_GZ in CI and improve documentation

### PR DESCRIPTION
**Pull request by traversaro**
As in the meanwhile we also added support for Ubuntu 24.04 (see https://github.com/robotology/robotology-superbuild/pull/1642), I also document that at the moment you can't install gz-sim with apt on Ubuntu 24.04 (see https://github.com/gazebosim/gz-sim/issues/2390#issuecomment-2096728527).

### CODE REVIEW

**Comment by traversaro** [comment]
The only failures are due to https://github.com/robotology/robotology-superbuild/issues/1646 that is not specific to this PR and will be solved by https://github.com/robotology/blocktest-yarp-plugins/pull/16, so the PR is ready for review.

**Review by Nicogene — APPROVED** [review]

File: doc/cmake-options.md [hunk 1]
```diff
@@ -91,21 +91,23 @@ Not all options are supported on all platforms. The following table provides a r
...
            
 
 <b id="f1">1!</b>:`ROBOTOLOGY_USES_MUJOCO` does not support building with apt dependencies on Debian or Ubuntu distributions older than 2022. Furthermore, it does not support build on Windows with Visual Studio 2019, it requires Visual Studio 2022.
+
 <b id="f2">2!</b>:`ROBOTOLOGY_ENABLE_ROBOT_TESTING` and `ROBOTOLOGY_ENABLE_ICUB_HEAD` do not support building with conda-forge dependencies on Apple Silicon.
-<b id="f3">3!</b>:`ROBOTOLOGY_USES_OCTAVE` do not support building with apt dependencies on Ubuntu 20.04.
+<b id="f3">3!</b>:`ROBOTOLOGY_USES_GZ_SIM`  with apt dependencies do not support building on Debian distros (only Ubuntu is supported). At the moment Ubuntu 24.04 apt packages are not available (https://github.com/gazebosim/gz-sim/issues/2390#issuecomment-2096728527). Furthermore it does not run on Windows (https://github.com/gazebosim/gz-sim/issues/2089) and have known problems on macOS (https://github.com/robotology/gz-sim-yarp-plugins/issues/90).
```
Comment (xela-95) [inline doc/cmake-options.md:109]:
```suggestion

<b id="f3">3!</b>:`ROBOTOLOGY_USES_GZ_SIM`  with apt dependencies do not support building on Debian distros (only Ubuntu is supported). At the moment Ubuntu 24.04 apt packages are not available (https://github.com/gazebosim/gz-sim/issues/2390#issuecomment-2096728527). Furthermore it does not run on Windows (https://github.com/gazebosim/gz-sim/issues/2089) and have known problems on macOS (https://github.com/robotology/gz-sim-yarp-plugins/issues/90).
```

File: doc/cmake-options.md [hunk 2]
```diff
@@ -91,21 +91,23 @@ Not all options are supported on all platforms. The following table provides a r
...
+<b id="f4">4!</b>:`ROBOTOLOGY_USES_OCTAVE` do not support building with apt dependencies on Ubuntu 20.04.
```
Comment (xela-95) [inline doc/cmake-options.md:110]:
```suggestion

<b id="f4">4!</b>:`ROBOTOLOGY_USES_OCTAVE` do not support building with apt dependencies on Ubuntu 20.04.
```

**Review by xela-95 — APPROVED** [review]
Only minor changes on formatting

LGTM! 🚀

**Pull request merged**